WWCodeTokyo/speak-her-db
# feat: adds pagination component

References: 
https://github.com/WWCodeTokyo/speak-her-db/issues/124

**Proposed Changes**
Paginate the list of speakers on the home page of speakher. For now I just put 50 statically. I think with airtable it's difficult to adjust the page size without reloading the whole thing because of the way `eachPage` of airtablejs works. If the db layer is switched in the future this component could be extended with a small dropdown maybe to allow for page size selection.

As of now there is also no way to receive a count of the records so I couldn't add a `... of 60` or `... of many` to this component. Could be a future improvement as well if the db layer changes away from airtable.

![image](https://user-images.githubusercontent.com/18051009/94932078-8b987f80-0503-11eb-8320-b3217592af9d.png)
![image](https://user-images.githubusercontent.com/18051009/94932109-918e6080-0503-11eb-8b10-7be878e84c4c.png)

*Description of changes in this PR*
Adds a pagination component that can be re-used on other views.
Adds this pagination component and pagination handling to the `FindSpeaker` component.

**Testing Plan**
I didn't see a test setup yet so I kept the PR without any tests. Let me know if you want me to write any unit or GUI tests.

*How do the reviewers QA this? Feel free to write unit tests or describe manual testing steps here*
- Open the start page
- Use the navigation on the start page.

**Known issues**
Sadly, due to how airtablejs handles pagination within `eachPage`, the `next page` button is active even when there is no more entries anymore. This behavior is gone, once the user presses the `next page` button again as we switch to internal pagination logic.
